src/main/java/software/aws/neptune/common/gremlindatamodel/resultset/ResultSetGetTypeInfo.java

```diff
@@ -0,0 +1,156 @@
+        COLUMN_TYPE_MAP.put("SEARCHABLE", Short.class);
+        COLUMN_TYPE_MAP.put("UNSIGNED_ATTRIBUTE", Boolean.class);
+        COLUMN_TYPE_MAP.put("FIXED_PREC_SCALE", Boolean.class);
+        COLUMN_TYPE_MAP.put("AUTO_INCREMENT", Boolean.class);
+        COLUMN_TYPE_MAP.put("LOCAL_TYPE_NAME", String.class);
+        COLUMN_TYPE_MAP.put("MINIMUM_SCALE", Short.class);
+        COLUMN_TYPE_MAP.put("MAXIMUM_SCALE", Short.class);
+        COLUMN_TYPE_MAP.put("SQL_DATA_TYPE", Integer.class);
+        COLUMN_TYPE_MAP.put("SQL_DATETIME_SUB", Integer.class);
+        COLUMN_TYPE_MAP.put("NUM_PREC_RADIX", Integer.class);
+    }
+
+    protected static void populateConstants(List<Map<String, Object>> typeInfo) {
+        for (Map<String, Object> info : typeInfo) {
+            info.put("CREATE_PARAMS", null);
+            info.put("NULLABLE", typeNullable);
+            info.put("SEARCHABLE", typeSearchable);
+            info.putIfAbsent("UNSIGNED_ATTRIBUTE", false);
+            info.put("FIXED_PREC_SCALE", false);
+            info.put("AUTO_INCREMENT", false);
+            info.put("LOCAL_TYPE_NAME", null);
+            info.put("SQL_DATA_TYPE", null);
+            info.put("SQL_DATETIME_SUB", null);
+
+            // Null for non-numerics
+            info.putIfAbsent("MINIMUM_SCALE", null);
+            info.putIfAbsent("MAXIMUM_SCALE", null);
+            info.putIfAbsent("NUM_PREC_RADIX", null);
+        }
+    }
+
+    public ResultSetGetTypeInfo(Statement statement, List<Map<String, Object>> rows) {
+        super(statement, ORDERED_COLUMNS, rows.size());
+        this.typeInformation = rows;
+    }
+
+    @Override
+    public Object getConvertedValue(final int columnIndex) throws SQLException {
+        verifyOpen();
+        final int index = getRowIndex();
+        if ((index < 0) || (index >= this.typeInformation.size())) {
+            throw SqlError
+                    .createSQLException(LOGGER, SqlState.DATA_EXCEPTION, SqlError.INVALID_INDEX, getRowIndex() + 1,
+                            this.typeInformation.size());
+        }
+        if ((columnIndex <= 0) || (columnIndex > ORDERED_COLUMNS.size())) {
+            throw SqlError
+                    .createSQLException(LOGGER, SqlState.DATA_EXCEPTION, SqlError.INVALID_COLUMN_INDEX, columnIndex,
+                            ORDERED_COLUMNS.size());
+        }
+
+        final String key = ORDERED_COLUMNS.get(columnIndex - 1);
+        if (this.typeInformation.get(index).containsKey(key)) {
+            final Object data = this.typeInformation.get(index).get(key);
+            this.wasNull = (data == null);
+            return data;
+        } else {
+            throw SqlError.createSQLFeatureNotSupportedException(LOGGER);
+        }
+    }
+
+    @Override
+    protected ResultSetMetaData getResultMetadata() {
+        final List<Class<?>> rowTypes = new ArrayList<>();
+        for (final String column : ORDERED_COLUMNS) {
```

Comment:
You could make this a bit more succinct (if you wanted) by using streams:
`ORDERED_COLUMNS.stream().map(e -> COLUMN_TYPE_MAP.get(e)).collect(Collectors.toList())`

Reply:
I would prefer not to since I don't see value in making code styling inconsistent within the project for something that arguably is only a styling benefit, also adding a layer of knowledge required for readability.

Reply:
That's fine.